### PR TITLE
armadactl refactor and testing

### PR DESCRIPTION
**Pull request by severinson**
This PR moves the business logic of armadactl into `internal/armadactl` and introduces a way to easily capture its output to facilitate automated testing. This PR also introduces several automated tests for armadactl. For these tests, armadactl is run against an Armada server that is automatically spun up in a Docker container.

### CODE REVIEW

File: cmd/armadactl/cmd/queue_test.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+		GroupOwners    []string
+		ResourceLimits map[string]float64
+		err            error // expected error, or nil if no error is expected
+	}{
+		"default flags":         {nil, nil, nil, nil, nil, nil},
+		"valid priority":        {[]flag{{"priorityFactor", "0.1"}}, makeFloat64Pointer(0.1), nil, nil, nil, nil},
+		"valid owners":          {[]flag{{"owners", "user1,user2"}}, nil, []string{"user1", "user2"}, nil, nil, nil},
+		"valid group owners":    {[]flag{{"groupOwners", "group1,group2"}}, nil, nil, []string{"group1", "group2"}, nil, nil},
+		"valid resource limits": {[]flag{{"resourceLimits", "cpu=0.3,memory=0.2"}}, nil, nil, nil, map[string]float64{"cpu": 0.3, "memory": 0.2}, nil},
+		// "invalid priority":        {[]flag{flag{"priorityFactor", "not_a_float"}}, makeFloat64Pointer(0.1), nil, nil, nil, nil},
```
Comment (carlocamurri) [inline cmd/armadactl/cmd/queue_test.go:53]: leave out of this PR.

File: cmd/armadactl/main.go [hunk 2]
```diff
@@ -5,7 +5,25 @@ import (
...
 )
 
+// Config is handled by cmd/params.go
 func main() {
-	common.ConfigureCommandLineLogging()
+	common.ConfigureCommandLineLogging() // TODO: can be removed once all of armadactl prints to its internal io.Writer
 	cmd.Execute()
 }
+
+// package main
```
Comment (carlocamurri) [inline cmd/armadactl/main.go:14]: Remove this

File: cmd/armadactl/cmd/watch.go [hunk 3]
```diff
@@ -1,88 +1,45 @@
...
-	summary += fmt.Sprintf(" | %s, job id: %s", reflect.TypeOf(e).String()[5:], e.GetJobId())
-
-	if kubernetesEvent, ok := e.(api.KubernetesEvent); ok {
-		summary += fmt.Sprintf(" pod: %d", kubernetesEvent.GetPodNumber())
+func watchCmd() *cobra.Command {
+	a := armadactl.New()
+	cmd := &cobra.Command{
+		Use:   "watch <queue> <jobSet>",
+		Short: "Watch job events in job set.",
+		Long:  `This command will list all job set events and `,
```
Comment (d80tb7) [inline cmd/armadactl/cmd/watch.go:20]: and...?!

Comment (severinson) [reply cmd/armadactl/cmd/watch.go:20]: This is what it said originally. I must've forgotten to fix it. Do you know what it's supposed to say? I don't have a good understanding of what the watch command is supposed to be doing.

**Review by d80tb7 — DISMISSED** [review]

**Review by d80tb7 — APPROVED** [review]

**Review by carlocamurri — APPROVED** [review]

**Pull request merged**